httomo/sweep_runner/param_sweep_runner.py

```diff
@@ -124,22 +130,59 @@ def determine_stages(self) -> Stages:
             after_sweep=NonSweepStage(after_sweep),
         )
 
+    def modify_loader_preview(self):
+        """
+        Modifies loader's preview based on the method present in the pipeline.
+        In some cases we need to override users given preview, e.g., when Paganin filter is present.
+        """
+
+        # we call this only to find the global size of the data (not very efficient)
+        source = self._pipeline.loader.make_data_source(padding=(0, 0))
+
+        # before modifying preview here we need to check if the block fits the memory if Paganin method is present in the pipeline
+        for method in self._pipeline._methods:
+            if "paganin" in method.method_name and method.sweep:
+                # Specifically dealing with the Paganin filter variable kernel size, as if the kernel is large,
+                # the larger preview needs to be taken in that case. So far this is the only method that
+                # requires an extended preview.
+
+                vertical_slices_preview_Paganin = paganin_kernel_estimator(
+                    method.config_params["pixel_size"],
+                    method.config_params["alpha"],
+                    method.config_params["energy"],
+                    method.config_params["dist"],
+                    vert_min_limit=self._vertical_slices_preview,
+                    peak_height=0.01,
+                )
+                self._vertical_slices_preview = np.max(vertical_slices_preview_Paganin)
+
+                vertical_slices_preview_max = _slices_to_fit_memory_Paganin(source)
+
+                # we do not stop the run if the Paganin's kernel is too wide, but extend the preview to the maximum allowed size for the current card
+                if self._vertical_slices_preview > vertical_slices_preview_max:
+                    self._vertical_slices_preview = vertical_slices_preview_max
+
+        preview_new_start_stop = _preview_modifier(
+            self._pipeline.loader.preview,
+            source.raw_shape[1],
+            self._vertical_slices_preview,
+        )
+
+        self._pipeline.loader.preview = PreviewConfig(
+            angles=self._pipeline.loader.preview.angles,
+            detector_y=PreviewDimConfig(
+                start=preview_new_start_stop[0], stop=preview_new_start_stop[1]
+            ),
+            detector_x=self._pipeline.loader.preview.detector_x,
+        )
```

Comment:
Currently I get a type error from pyright telling me:
```
Cannot assign to attribute "preview" for class "LoaderInterface"
  "PreviewConfig" is not assignable to "property"
```

The reason for this error is that, while a setter for preview has been added to `StandardLoaderWrapper`, a setter for preview hasn't been added to the `LoaderInterface` protocol. In this `modify_loader_preview()` method, all we know about `self._pipeline.loader` is that it's an object that implements `LoaderInterface`, and that protocol has no preview setter, so the type checker correctly flags this as a type error.

Could you fix this by adding a preview setter to `LoaderInterface`? (Pyright/mypy should stop complaining once that's done).

Reply:
Sorry cannot see the error with `mypy` somehow, can you please give it a go?

Reply:
No worries, added in 516f7fb